src/util/psbt/serialize.rs

```diff
@@ -355,7 +355,7 @@ impl Deserialize for TapTree {
             builder = builder.add_leaf_with_ver(*depth, script, leaf_version)
                 .map_err(|_| encode::Error::ParseFailed("Tree not in DFS order"))?;
         }
-        if builder.is_finalized() {
+        if builder.is_finalized() || !builder.has_hidden_nodes() {
```

Comment:
In 77715311cf0a68a3623f989393d4a58bc7e884ba:

I don't understand the name `is_finalized` but I'm pretty sure this `||` is supposed to be a `&&`.

Reply:
This is a serious issue.